In [10]:
import numpy as np
import pandas as pd
from pydataset import data
import os
import acquire as acq
import env
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
import seaborn as sns

Attendance Data

Read the data from the attendance table and calculate an attendance percentage for each student. One half day is worth 50% of a full day, and 10 tardies is equal to one absence.

In [32]:
attendance = pd.read_clipboard()
attendance

student_id 1/1/18 1/2/18 1/3/18 1/4/18 1/5/18 1/6/18 1/7/18 1/8/18
0      Sally      P      T      T      H      P      A      T      T
1       Jane      A      P      T      T      T      T      A      T
2      Billy      A      T      A      A      H      T      P      T
3       John      P      T      H      P      P      T      P      P

In [38]:
attendance.student_id

0    Sally
1     Jane
2    Billy
3     John
Name: student_id, dtype: object

In [30]:
days_by_student = attendance.student[''].count()

KeyError: 'Column not found: status'

In [ ]:
tardies_by_student = attendance_by_student['status'].apply(lambda x: (x == 'T').sum())

In [ ]:
half_days_by_student = attendance_by_student['status'].apply(lambda x: (x == 'H').sum() / 2)

In [ ]:
attendance_percentages = (days_by_student + half_days_by_student * 0.5 - tardies_by_student / 10) / days_by_student * 100


print(attendance_percentages)


Coffee Levels

Read the coffee_levels table.


In [15]:
coffee = pd.read_clipboard()
coffee

hour coffee_carafe  coffee_amount
0      8             x       0.816164
1      9             x       0.451018
2     10             x       0.843279
3     11             x       0.335533
4     12             x       0.898291
5     13             x       0.310711
6     14             x       0.507288
7     15             x       0.215043
8     16             x       0.183891
9     17             x       0.391560
10     8             y       0.189297
11     9             y       0.521502
12    10             y       0.023163
13    11             y       0.235529
14    12             y       0.017009
15    13             y       0.997464
16    14             y       0.058361
17    15             y       0.144644
18    16             y       0.544676
19    17             y       0.594126
20     8             z       0.999264
21     9             z       0.915990
22    10             z       0.144928
23    11             z       0.311495
24    12             z       0.771947
25    13             z       0.398520
26    14             z       0.864464
27    15             z       0.436364
28    16             z       0.280621
29    17             z       0.436677

Transform the data so that each carafe is in it's own column.


Is this the best shape for the data?

Cake Recipes

Read the cake_recipes table. This data set contains cake tastiness scores for combinations of different recipes, oven rack positions, and oven temperatures.


In [37]:
recipe = pd.read_clipboard()
recipe

recipe:position        225        250        275        300
0        a:bottom  61.738655  53.912627  74.414730  98.786784
1           a:top  51.709751  52.009735  68.576859  50.228470
2        b:bottom  57.095320  61.904369  61.196980  99.248541
3           b:top  82.455004  95.224151  98.594881  58.169350
4        c:bottom  96.470207  52.001358  92.893227  65.473084
5           c:top  71.306308  82.795477  92.098050  53.960273
6        d:bottom  52.799753  58.670419  51.747686  56.183110
7           d:top  96.873178  76.101363  59.571620  50.971626

Tidy the data as necessary.


Which recipe, on average, is the best? recipe b


Which oven temperature, on average, produces the best results? 275


Which combination of recipe, rack position, and temperature gives the best result? recipe b, bottom rack, 300 degrees